Assembler.Base/DefaultAssemblerFactory.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System.Collections.Generic;
+using Assembler.Base.Creators;
+using Assembler.Base.FrameHandlers;
+using Assembler.Base.MessageEnrichers;
+using Assembler.Base.Releasing;
+using Assembler.Core;
+using Assembler.Core.Entities;
+using Assembler.Core.Enums;
+using Assembler.Core.RawAssemblingEntities;
+using Microsoft.Extensions.Logging;
+
+namespace Assembler.Base
+{
+    public class DefaultAssemblerFactory<TFrame, TMessageInAssembly>
+        where TFrame : BaseFrame
+        where TMessageInAssembly : BaseMessageInAssembly
+    {
+        // TODO: Make it a builder after the CR
+        public IAssembler<TFrame> Create(
+            ITimeBasedCache<TMessageInAssembly> timeBasedCache,
+            IIdentifierGenerator<TFrame> identifierGenerator,
+            IMessageInAssemblyCreator<TMessageInAssembly> messageInAssemblyCreator,
+            IMessageEnricher<TFrame, TMessageInAssembly> initialFrameMessageEnricher,
+            IMessageEnricher<TFrame, TMessageInAssembly> middleFrameMessageEnricher,
+            IMessageEnricher<TFrame, TMessageInAssembly> finalFrameMessageEnricher,
+            IMessageReleaser<TMessageInAssembly> messageInAssemblyReleaser, ILoggerFactory loggerFactory)
+        {
+            var dateTimeProvider = new DateTimeProvider();
+
+            var initialFrameHandler = new InitialFrameHandler<TFrame, TMessageInAssembly>(timeBasedCache,
+                identifierGenerator, messageInAssemblyCreator, initialFrameMessageEnricher, messageInAssemblyReleaser,
+                dateTimeProvider, loggerFactory);
+            var middleFrameHandler = new MiddleFrameHandler<TFrame, TMessageInAssembly>(timeBasedCache,
+                identifierGenerator, messageInAssemblyCreator, middleFrameMessageEnricher, messageInAssemblyReleaser,
+                dateTimeProvider, loggerFactory);
+            var finalFrameHandler = new FinalFrameHandler<TFrame, TMessageInAssembly>(timeBasedCache,
+                identifierGenerator, messageInAssemblyCreator, finalFrameMessageEnricher, messageInAssemblyReleaser,
+                dateTimeProvider, loggerFactory);
+
+            var resolver = new GeneralResolver<AssemblingPosition, IFrameHandler<TFrame>>(
+                new Dictionary<AssemblingPosition, IFrameHandler<TFrame>>
+            {
+                {AssemblingPosition.Initial, initialFrameHandler },
+                {AssemblingPosition.Middle, middleFrameHandler },
+                {AssemblingPosition.Final, finalFrameHandler }
+            });
+
+            messageInAssemblyReleaser.Start();
+
+            return new FramesAssembler<TFrame>(resolver, loggerFactory);
+        }
+
+        public IAssembler<TFrame> CreateRawAssembler(ITimeBasedCache<RawMessageInAssembly> timeBasedCache,
+            IIdentifierGenerator<TFrame> identifierGenerator, ILoggerFactory loggerFactory,
+            IMessageReleaser<RawMessageInAssembly> messageInAssemblyReleaser = null)
```

Comment:
I think its important to explain the user that he can use the ConditionalMessageReleaser with the Validator (such as composite).

What do you think is the best way to do this? Summary with explanation to this argument in this "CreateRawAssembler" method? **maybe in the ReadME**? maybe another function?

I think the ReadME is the best, because this note to the user (that he can use the conditional) is important and relevant also if he uses `Create` Method...

Reply:
Readme, but I'd also add a summary for the builder